docs/library/rp2.pio.rst

```diff
@@ -0,0 +1,164 @@
+.. currentmodule:: rp2
```

Comment:
It seemed like a good idea to put all the PIO-related stuff in its own file. This doesn't strictly fit the module structure--both the Flash and PIO stuff are scoped under `rp2`--so I wasn't exactly sure how best to do that.